server/src/main/java/org/elasticsearch/index/translog/MultiSnapshot.java

```diff
@@ -73,6 +80,7 @@ public int overriddenOperations() {
                     overriddenOperations++;
                 }
             }
+            skippedOperations += current.skippedOperations();
```

Comment:
why maintain the state here? you can also ask the underlying snapshots when people call `skippedOperations`? this will also give you accurate counting - right now we only have visibility into the skipped counters once a generation is fully read - I think this is confusing. This too smells like a unit test coverage issue.

Reply:
agreed, test is there - IndexLevelReplicationTests#testSeqNoCollision